modules/greenbidsRtdProvider.js

```diff
@@ -16,22 +17,28 @@ function init(moduleConfig) {
     return false;
   } else {
     rtdOptions.pbuid = params?.pbuid;
-    rtdOptions.targetTPR = params?.targetTPR || 0.99;
     rtdOptions.timeout = params?.timeout || 200;
     return true;
   }
 }
 
 function onAuctionInitEvent(auctionDetails) {
-  auctionInfo.auctionId = auctionDetails.auctionId;
+  /* Emitting one billing event per auction */
+  events.emit(CONSTANTS.EVENTS.BILLABLE_EVENT, {
+    type: 'auction',
+    billingId: generateUUID(),
+    auctionId: auctionDetails.auctionId,
+    vendor: MODULE_NAME
+  });
```

Comment:
So the plan is to emit a billing event once per auction no matter what right, even if the adapter did not do something?

I am not aware of if there is a scenario where that would happen, like if the greenbids endpoint returns empty or errors.

Just checking.

Reply:
Yes I think billing for each auction makes the most sense because it correlates 1 to 1 with our infra cost, but I updated this to check the that the `greenbidsId` is set in the ortb2Imp.ext this way we won't bill if the call to greenbids wasn't successful. Updates the tests to reflect this behavior.